Mail2Bug/Email/AckEmailHandler.cs

```diff
@@ -28,21 +30,26 @@ public void SendAckEmail(IIncomingEmailMessage originalMessage, string workItemI
             var ewsMessage = originalMessage as EWSIncomingMessage;
             if (ewsMessage != null)
             {
-                HandleEWSMessage(ewsMessage, workItemId);
+                HandleEWSMessage(ewsMessage, workItem);
             }
         }
 
-        private void HandleEWSMessage(EWSIncomingMessage originalMessage, string workItemId)
+        private void HandleEWSMessage(EWSIncomingMessage originalMessage, SimpleWorkItem workItem)
         {
-            originalMessage.Reply(GetReplyContents(workItemId), _config.EmailSettings.AckEmailsRecipientsAll);
+            originalMessage.Reply(GetReplyContents(workItem), _config.EmailSettings.AckEmailsRecipientsAll);
         }
 
-        private string GetReplyContents(string workItemId)
+        private string GetReplyContents(SimpleWorkItem workItem)
         {
             var bodyBuilder = new StringBuilder();
             bodyBuilder.Append(_config.EmailSettings.GetReplyTemplate());
-            bodyBuilder.Replace("[BUGID]", workItemId);
```

Comment:
With the new support you've created, I think this can be both simplified and made more powerful:
1. Replace the special handling of work-item fields by just providing the syntax [FieldName:default value] (or some similar syntax), where this syntax would be handled by fetching the specified field from the item, and if it fails (returns 'null') the default would be used
2. For the non-work-item based replacements, you can keep them as is